src/Controllers/ResultController.php

```diff
@@ -6,55 +6,62 @@
 use Framework\Http\Response;
 use Psr\Http\Message\RequestInterface;
 use QuizApp\Services\AbstractService;
-use ReallyOrm\Exceptions\NoSuchRowException;
+use QuizApp\Utils\PaginatorTrait;
 
 /**
  * Class ResultController
  * @package QuizApp\Controllers
  */
-//TODO local.quiz.com/results?page=1
 class ResultController extends AbstractController
 {
-    private $resultService;
-
-    public function setService(AbstractService $resultService)
+    use PaginatorTrait;
+    /**
+     * @var AbstractService
+     */
+    protected $resultService;
+    /**
+     * This function sets the resultService
+     * @param AbstractService $service
+     */
```

Comment:
Why do you need this setter? Do you have setter injection?

Reply:
We have setter injection called from container, different controllers have different services dependency